### ISSUE TITLE
Moving model to gpu ignores `NoGradArray`

### ISSUE THREAD

**Issue by nmheim** [nmheim/GenerativeModels.jl#44]
With Zygote and our custom [`NoGradArray`](https://github.com/nmheim/GenerativeModels.jl/blob/julia-1.3/src/utils/nogradarray.jl) on the `julia-1.3` branch
```julia
p = Gaussian(zeros(3), NoGradArray(ones(3)))
params(p)  # returns Params([[0.0, 0.0, 0.0]])

g = gpu(p)
params(g) # returns Params([Float32[0.0, 0.0, 0.0], Float32[1.0, 1.0, 1.0]])
```

This happens because:
```julia
x = NoGradArray(zeros(3))
gpu(x)  # returns a normal CuArray
```

**Comment by nmheim** [comment]
the same is true for save/load of models that contain `NoGradArray`s. they are ignored and become normal `Array`s. this also happens with `StaticArrays`...

**Comment by nmheim** [comment]
This also breaks the Wasserstein loss on the GPU...

**Comment by nmheim** [comment]
fixed in #48
